### PR TITLE
Circuit: Add tests for orchard circuit

### PR DESCRIPTION
**Pull request by ConstanceBeguier**
Add some positive and negative tests for Orchard circuit

### CODE REVIEW

File: src/circuit.rs [hunk 1]
```diff
@@ -1404,4 +1407,235 @@ mod tests {
...
+            AssetBase::random(&mut rng),
+            pallas::Base::random(&mut rng),
+            pallas::Base::random(&mut rng),
+            NoteCommitTrapdoor(pallas::Scalar::random(&mut rng)),
+        )
+        .unwrap()
+    }
+
+    #[test]
+    fn test_orchard_circuit() {
```
Comment (PaulLaux) [inline src/circuit.rs:1548]: Let's call this `orchard_circuit_negative_test()`

**Review by PaulLaux — APPROVED** [review]
Approved with minor comment

File: src/circuit.rs [hunk 2]
```diff
@@ -1441,4 +1445,232 @@ mod tests {
...
+            )
+        };
+
+        let cmx = output_note.commitment().into();
+
+        let value = if split_flag {
+            NoteValue::zero() - output_note.value()
+        } else {
+            spent_note.value() - output_note.value()
+        };
```
Comment (PaulLaux) [inline src/circuit.rs:1520]: Let's unify this part with the previous `if split_flag`. Plus a comment to describe rationale.

File: src/circuit.rs [hunk 3]
```diff
@@ -1441,4 +1445,232 @@ mod tests {
...
+                nf_old,
+                rk,
+                cmx,
+                enable_spend: true,
+                enable_output: true,
+            },
+        )
+    }
+
+    fn check_proof_of_orchard_circuit(circuit: &Circuit, instance: &Instance, should_pass: bool) {
```
Comment (PaulLaux) [inline src/circuit.rs:1550]: Let's put this function above `generate_circuit_instance()`

File: src/circuit.rs [hunk 4]
```diff
@@ -1441,4 +1445,232 @@ mod tests {
...
+    #[test]
+    fn orchard_circuit_negative_test() {
+        let mut rng = OsRng;
+
+        for is_native_asset in [true, false] {
+            for split_flag in [true, false] {
+                let (circuit, instance) =
+                    generate_circuit_instance(is_native_asset, split_flag, &mut rng);
+
+                check_proof_of_orchard_circuit(&circuit, &instance, true);
```
Comment (PaulLaux) [inline src/circuit.rs:1591]: Very nice.

**Review by PaulLaux — APPROVED** [review]
Approved with minor comments.

File: src/circuit.rs (hunk 2 above)
Comment (ConstanceBeguier) [reply src/circuit.rs:1520]: Done

File: src/circuit.rs (hunk 3 above)
Comment (ConstanceBeguier) [reply src/circuit.rs:1550]: Done

**Pull request merged**